src/Persistence/PersistentObjectFactory.php

```diff
@@ -449,12 +449,6 @@ protected function normalizeObject(string $field, object $object): object
 
     final protected function initializeInternal(): static
     {
-        if (!Configuration::isBooted()) {
-            return $this;
-        }
-
-        $this->persist = $this->isPersistenceEnabled() ? PersistMode::PERSIST : PersistMode::WITHOUT_PERSISTING;
```

Comment:
this is a leftover from an old dev, now this is not needed. And the `Configuration::isBooted()` before was preventing to add the `afterInstanntiate` hook (in the case when the phpunit extension is not used)